pkg/apis/pingcap/v1alpha1/tiflash_config.go

```diff
@@ -32,9 +32,9 @@ type TiFlashConfig struct {
 // +k8s:openapi-gen=false
 type FlashServerConfig struct {
 	// +optional
-	EngineAddr string `json:"engine-addr,omitempty" toml:"engine-addr,omitempty"`
+	EngineAddr *string `json:"engine-addr,omitempty" toml:"engine-addr,omitempty"`
```

Comment:
Please also update the default config logic in https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tiflash_util.go#L122.
And also make sure that TiFlash can start up successfully, it requires some configs to exist even if they are empty strings.

Reply:
updated.